src/shogun/kernel/PeriodicKernel.h

```diff
@@ -0,0 +1,180 @@
+/*
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Written (W) 2015 Esben Soerig
+ */
+
+#ifndef PERIODICKERNEL_H
+#define PERIODICKERNEL_H
+
+#include <shogun/lib/config.h>
```

Comment:
Most of those general includes do not need to be there actually, try removing most of them. The base class should theoretically do it